resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverBuilder.java

```diff
@@ -104,6 +105,17 @@ public DnsNameResolverBuilder nameServerAddresses(DnsServerAddresses nameServerA
     }
 
     /**
+     * Sets the cache for resolution results.
+     *
+     * @param resolveCache the DNS resolution results cache
+     * @return {@code this}
+     */
+    public DnsNameResolverBuilder resolveCache(DnsCache resolveCache) {
+        this.resolveCache  = resolveCache;
```

Comment:
Could you:

- modify min/max/negativeTtl() methods to raise an IllegalStateException when a user called `resolveCache()` already?
- modify this method to raise an IllegalStateException when a user called min/max/negativeTtl() methods already?

.. because they are mutually exclusive.